### PR TITLE
fix(dataresolver): ensure fetched file is convert to a buffer

### PR DESCRIPTION
**Pull request by nev-r**
**Please describe the changes this PR makes and why it should be merged:**
creating an emoji using a web URL is currently nonfunctional in `main` but working in `stable`.

`stable`->`main` removed `resolveFileAsBuffer` but `DataResolver.resolveBase64` still expected a Buffer,
and `resolveFile` provided a ReadableStream.

there's no reason to readd/reimplement that function since it only acted as an intermediary between those 2 other static methods, in this one specific case. so now `resolveFile` will directly return a Buffer, as the typing suggests it's supposed to.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/discord.js/src/util/DataResolver.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ class DataResolver extends null {
       return image;
     }
     const file = await this.resolveFile(image);
-    return DataResolver.resolveBase64(file);
+    return this.resolveBase64(file);
```
Comment (almeidx) [inline packages/discord.js/src/util/DataResolver.js:70]: The method is static
```suggestion
    return DataResolver.resolveBase64(file);
```

Comment (nev-r) [reply packages/discord.js/src/util/DataResolver.js:70]: hm, just standardizing across the file. every other other static->static reference is using `this`. should the rest of the file be swapped over the other way then? as DataResolver is just a container for static methods, there isn't really a binding problem here.

Comment (DTrombett) [reply packages/discord.js/src/util/DataResolver.js:70]: > The method is static
> ```suggestion
>     return DataResolver.resolveBase64(file);
> ```

`this` is `DataResolver` here so no need of this change...

Comment (almeidx) [reply packages/discord.js/src/util/DataResolver.js:70]: > just standardizing across the file. every other static->static reference is using `this`

Didn't notice the rest of the static methods were using `this`, you can ignore this suggestion 

> should the rest of the file be swapped over the other way then

IMO, yes, but that's not up to me

**Review by SpaceEEC — APPROVED** [review]
LGTM

I went ahead and also applied this fix to the part resolving a local file path. 😄

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**